prefect_gcp/cloud_run_job.py

```diff
@@ -0,0 +1,686 @@
+            #     r"Resource '[^ ]+' of kind 'JOB' in region '[\w\-0-9]+' "
+            #     r"in project '[\w\-0-9]+' does not exist"
+            # )
+            if re.findall(pat1, str(exc)):
+                raise RuntimeError(
+                    f"Failed to find resources at {exc.uri}. "
+                    "Confirm that region '{self.region}' is "
+                    "the correct region for your Cloud Run Job "
+                    "and that '{self.credentials.project_id}' is the "
+                    "correct GCP project. If your project ID is not "
+                    "correct, you are using a Credentials "
+                    "block with permissions for the wrong project."
+                ) from exc
+            else:
+                raise exc
+
+        raise exc
+
+    @sync_compatible
+    async def run(self, task_status: Optional[TaskStatus] = None):
+        """Run the configured job on a Google Cloud Run Job."""
+        with self._get_client() as client:
+            await run_sync_in_worker_thread(
+                self._create_job_and_wait_for_registration, client
+            )
+            job_execution = await run_sync_in_worker_thread(
+                self._begin_job_execution, client
+            )
+
+            if task_status:
+                task_status.started(self.job_name)
+
+            result = await run_sync_in_worker_thread(
+                self._watch_job_execution_and_get_result,
+                client,
+                job_execution,
+                5,
+            )
+            return result
+
+    def _create_job_and_wait_for_registration(self, client: Resource) -> None:
+        """Create a new job wait for it to finish registering."""
+        try:
+            self.logger.info(f"Creating Cloud Run Job {self.job_name}")
+            Job.create(
+                client=client,
+                namespace=self.credentials.project_id,
+                body=self._jobs_body(),
+            )
+        except googleapiclient.errors.HttpError as exc:
+            self._create_job_error(exc)
+
+        try:
+            self._wait_for_job_creation(client=client, timeout=self.timeout)
+        except Exception as exc:
+            self.logger.exception(
+                "Encountered an exception while waiting for job run creation"
+            )
+            if not self.keep_job:
+                self.logger.info(
+                    f"Deleting Cloud Run Job {self.job_name} from Google Cloud Run."
+                )
+                try:
+                    Job.delete(
+                        client=client,
+                        namespace=self.credentials.project_id,
+                        job_name=self.job_name,
+                    )
+                except Exception as exc:
+                    self.logger.exception(
+                        "Received an unexpected exception while attempting to delete"
+                        f" Cloud Run Job.'{self.job_name}':\n{exc!r}"
```

Comment:
`logger.exception` [includes the exception info by default](https://docs.python.org/3/library/logging.html#logging.exception).
```suggestion
                        f" Cloud Run Job.'{self.job_name}'"
```